app/src/main/java/org/fossasia/openevent/app/core/ticket/detail/TicketPrintAdapter.java

```diff
@@ -0,0 +1,120 @@
+package org.fossasia.openevent.app.core.ticket.detail;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.pdf.PdfDocument;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CancellationSignal;
+import android.os.ParcelFileDescriptor;
+import android.print.PageRange;
+import android.print.PrintAttributes;
+import android.print.PrintDocumentAdapter;
+import android.print.PrintDocumentInfo;
+import android.print.pdf.PrintedPdfDocument;
+import android.support.annotation.RequiresApi;
+
+import org.fossasia.openevent.app.data.ticket.Ticket;
+import org.fossasia.openevent.app.utils.DateUtils;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+@RequiresApi(api = Build.VERSION_CODES.KITKAT)
+public class TicketPrintAdapter extends PrintDocumentAdapter {
+
+    private final int TOTAL_PAGES = 1;
+    private final Context context;
+    private int pageHeight;
+    private int pageWidth;
+    private PdfDocument ticketDocument;
+    private final Ticket ticket;
+    Paint paintHeaderText = new Paint();
+    Paint paintTicketDescription = new Paint();
+
+    public TicketPrintAdapter(Context context, Ticket ticket) {
+        this.context = context;
+        this.ticket = ticket;
+    }
+
+    @Override
+    public void onLayout(PrintAttributes oldAttributes,
+                         PrintAttributes newAttributes,
+                         CancellationSignal cancellationSignal,
+                         LayoutResultCallback callback, Bundle extras) {
+        ticketDocument = new PrintedPdfDocument(context, newAttributes);
+        pageHeight =
+            newAttributes.getMediaSize().getHeightMils() / 1000 * 72;
+        pageWidth =
+            newAttributes.getMediaSize().getWidthMils() / 1000 * 72;
+
+        if (cancellationSignal.isCanceled()) {
+            callback.onLayoutCancelled();
+            return;
+        }
+
+        PrintDocumentInfo.Builder builder = new PrintDocumentInfo
+            .Builder("Ticket_" + ticket.event.getIdentifier() + "_" +ticket.getId() + ".pdf")
```

Comment:
ticket.getEvent() 

And this may fail on NullPointerException

Reply:
It is returning null

Reply:
Yes, event relation is lazy